File: spl-net/src/main/java/bgu/spl171/net/impl/TFTP/Commands/ReadRequest.java
package bgu.spl171.net.impl.TFTP.Commands;

import bgu.spl171.net.impl.TFTP.Commands.Responses.Response;

public class ReadRequest implements Command {

    String fileName;
    public ReadRequest(String fileName) {
        this.fileName = fileName;
    }

    public String getFileName() {
        return fileName;
    }

    @Override
    public short getOpCode() {
        return OpCode.ReadRequest.getValue();
    }

    @Override
    public Response execute() {
        return null;
    }
}

File: java to c++/ReadRequest.cpp
#include "ReadRequest.h"

namespace bgu
{
	namespace spl171
	{
		namespace net
		{
			namespace impl
			{
				namespace TFTP
				{
					namespace Commands
					{
						using Response = bgu::spl171::net::impl::TFTP::Commands::Responses::Response;

						ReadRequest::ReadRequest(const std::wstring &fileName)
						{
							this->fileName = fileName;
						}

						std::wstring ReadRequest::getFileName()
						{
							return fileName;
						}

						short ReadRequest::getOpCode()
						{
							return OpCode::ReadRequest::getValue();
						}
					}
				}
			}
		}
	}
}
